File: app/src/main/java/esa/mydemo/fragment/MyProfileViewModel.java
package esa.mydemo.fragment;

import android.content.Intent;
import android.view.View;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import esa.mydemo.main.LoginActivity;
import esa.mylibrary.info.UserInfo;

public class MyProfileViewModel extends ViewModel {
    // TODO: Implement the ViewModel

    //数据绑定
    public MutableLiveData<String> title = new MutableLiveData<>("个人信息");
    public MutableLiveData<String> userName = new MutableLiveData<>("用户名");
    public MutableLiveData<String> userId = new MutableLiveData<>("ID");
    public MutableLiveData<String> devInfo = new MutableLiveData<>("设备信息");
    public MutableLiveData<String> userInfo = new MutableLiveData<>("用户详细信息");
    private View view;
    private Fragment fragment;


    public void init(View view, Fragment myProfileFragment) {
        this.view = view;
        this.fragment = myProfileFragment;

    }

    /**
     * @param view
     * @return void
     * @description 注销
     * @author tvrddmss
     * @time 2023/4/7 21:18
     */
    public void onLogoutClick(View view) {
        UserInfo.setPassword("");

        //跳转页面
        Intent intent = new Intent(this.view.getContext(), LoginActivity.class);
        this.view.getContext().startActivity(intent);
        this.fragment.getActivity().finish();
    }


    /**
     * @param view
     * @return void
     * @description 修改密码
     * @author tvrddmss
     * @time 2023/4/7 21:18
     */
    public void onEditPasswordClick(View view) {

    }
}

File: app/src/main/java/esa/mydemo/fragment/myprofile/MyProfileViewModel.kt
package esa.mydemo.fragment.myprofile

import android.content.Intent
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.MutableLiveData
import esa.mydemo.base.BaseViewModel
import esa.mydemo.main.LoginActivity
import esa.mylibrary.info.UserInfo

class MyProfileViewModel : BaseViewModel() {
    // TODO: Implement the ViewModel
    //数据绑定
    var appversion = MutableLiveData("版本")
    var title = MutableLiveData("个人信息")
    var userName = MutableLiveData("用户名")
    var userId = MutableLiveData("ID")
    var devInfo = MutableLiveData("设备信息")
    var userInfo = MutableLiveData("用户详细信息")
    private lateinit var view: View
    private lateinit var fragment: Fragment
    fun init(view: View, myProfileFragment: Fragment) {
        this.view = view
        fragment = myProfileFragment
    }

    /**
     * @param view
     * @return void
     * @description 注销
     * @author tvrddmss
     * @time 2023/4/7 21:18
     */
    fun onLogoutClick(view: View?) {
        UserInfo.setPassword("")

        //跳转页面
        val intent = Intent(this.view!!.context, LoginActivity::class.java)
        this.view!!.context.startActivity(intent)
        fragment.activity?.finish()
    }

}